zebra-chain/src/amount.rs

```diff
@@ -191,6 +191,61 @@ where
     }
 }
 
+impl<C> PartialOrd for Amount<C>
+where
+    Amount<C>: Eq,
+    C: Constraint,
+{
+    fn partial_cmp(&self, other: &Amount<C>) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl<C> Ord for Amount<C>
+where
+    Amount<C>: Eq,
+    C: Constraint,
+{
+    fn cmp(&self, other: &Amount<C>) -> std::cmp::Ordering {
+        self.0.cmp(&other.0)
+    }
+}
```

Comment:
Another question:

Can we just derive PartialOrd and Ord on Amount?

Reply:
sounds good to me

Reply:
Actually if i try to implement `PartialEq` as suggested in the documentation i get a conflict with the one already at https://github.com/oxarbitrage/zebra/blob/block_subsidy/zebra-chain/src/amount.rs#L20

Reply:
Trait implementations need to be unique.

So we can either have derives for Eq, PartialEq, Ord, and PartialOrd, or implement them using code.

Reply:
I think I can make this happen.

Reply:
I made this change in c5a8d5a40c562e651a9d3e517b723fe9370d4fea in https://github.com/oxarbitrage/zebra/pull/29